backend/src/middleware/validation/validationMiddleware.js

```diff
@@ -61,31 +61,58 @@ const validateUpdatePost = async (resolve, root, args, context, info) => {
 
 const validateReport = async (resolve, root, args, context, info) => {
   const { resourceId } = args
-  const { user, driver } = context
+  const { user } = context
   if (resourceId === user.id) throw new Error('You cannot report yourself!')
+  return resolve(root, args, context, info)
+}
+
+const validateReview = async (resolve, root, args, context, info) => {
+  const { resourceId } = args
+  let existingReportedResource
+  const { user, driver } = context
+  if (resourceId === user.id) throw new Error('You cannot review yourself!')
   const session = driver.session()
-  try {
-    const reportQueryRes = await session.run(
+  const reportReadTxPromise = session.writeTransaction(async txc => {
+    const validateReviewTransactionResponse = await txc.run(
       `
-      MATCH (:User {id:$submitterId})-[:REPORTED]->(resource {id:$resourceId}) 
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-11-30T00:04:00Z" title="Saturday, November 30th 2019, 1:04:00 am +01:00">Nov 30, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -59,13 +59,23 @@ const validateUpdatePost = async (resolve, root, args, context, info) => {
 
 const validateReport = async (resolve, root, args, context, info) => {
   const { resourceId } = args
-  const { user, driver } = context
+  const { user } = context
   if (resourceId === user.id) throw new Error('You cannot report yourself!')
+  return resolve(root, args, context, info)
+}
+
+const validateReview = async (resolve, root, args, context, info) => {
+  const { resourceId } = args
+  const { user, driver } = context
+  if (resourceId === user.id) throw new Error('You cannot review yourself!')
   const session = driver.session()
   const reportQueryRes = await session.run(
     `
-      MATCH (:User {id:$submitterId})-[:REPORTED]->(resource {id:$resourceId}) 
-      RETURN labels(resource)[0] as label
+      MATCH (resource {id: $resourceId})
+      WHERE resource:User OR resource:Post OR resource:Comment
+      OPTIONAL MATCH (:User)-[filed:FILED]->(:Report {closed: false})-[:BELONGS_TO]->(resource)
```


Are there ever situations in which ~we have a `:Report` without a user who filed it or~ no `author:User` who `:WROTE` the resource?

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-11-30T09:47:35Z" title="Saturday, November 30th 2019, 10:47:35 am +01:00">Nov 30, 2019</time>_

---


I don't think so... I think this `OPTIONAL MATCH` is because the resource is not always a `Post` or `Comment` that was written by someone. If it is then we are checking that a moderator cannot review there own `Post|Comment`, see lines 100-105